### PR TITLE
fix(Core/Trainers): Added new column `ReqSpell` to `npc_trainer` table.

### LINKED ISSUE #5390: Goblin and Gnomish engineering accessible at same time
Originally reported https://github.com/chromiecraft/chromiecraft/issues/443

#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

As an engineering  225 you can learn both Goblin and Gnomish engineering at the same .

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Only one engineering option should be train-able at the time. having both goblin and gnomish engineering should not be possible.

https://wowpedia.fandom.com/wiki/Gnome_Engineering?oldid=2299320
https://wowpedia.fandom.com/wiki/Goblin_Engineering

> From patch 3.2.0 until Cataclysm, Gnomish Engineers and Goblin Engineers could switch profession specializations for 150g by visiting Narain Soothfancy's hut in northeast Tanaris and interacting with the book on the table. 

##### STEPS TO REPRODUCE THE PROBLEM:

1. Level a char to 20+ and a minimum of 205 engineering. 
2. .go c 387 - this gets you to Oglethorpe Obnoticus in Booty Bay, who is the gnomish eng trainer. Learn all his recipes.
3. .go c 23414 - this takes you to Nixx Sprocketspring, goblin eng trainer in Gadgetzan. He will also teach you all goblin recipes without any pre-req quests done
4. Now you have access to both sets:
WoWScrnShot_041921_212305

##### EXTRA NOTES:

Thanks for reproduction steps and extended discussion to @Azcobu and @necropola

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/b9ae587b6b37355448c30301f7cccd50175117cd

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/97973634-goblin-and-gnomish-engineering-accessible-at-same-time?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Prevent learning gnomish and goblin engineering recipes at same time.
Fixes #5390

## Issues Addressed:

- Closes #5390
- Closes https://github.com/chromiecraft/chromiecraft/issues/443

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Level a char to 20+ and a minimum of 205 engineering.
.go c 387 - this gets you to Oglethorpe Obnoticus in Booty Bay, who is the gnomish eng trainer. Learn all his recipes.
.go c 23414 - this takes you to Nixx Sprocketspring, goblin eng trainer in Gadgetzan. 
No goblin/gnomish recipes without gnomish/goblin engineering

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]
I think you can merge the sql files into one :)

File: data/sql/updates/pending_db_world/rev_1638616990639666200.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1638616990639666200');
+
+ALTER TABLE `npc_trainer` ADD COLUMN `ReqSpell` INT UNSIGNED DEFAULT 0 NOT NULL AFTER `ReqLevel`;
```
Comment (claudiodfc) [inline data/sql/updates/pending_db_world/rev_1638616990639666200.sql:3]: You can drop the other file and add this to this one
```
-- Gnomish Engineering
UPDATE `npc_trainer` SET `ReqSpell`= 20219 WHERE `SpellID` IN (12897, 30575, 30574, 12907, 56473, 12905, 30570, 12903, 23096, 30568, 15633, 23129, 11454, 12906, 12759, 12902, 30569, 12899, 23489, 36955, 11454);

-- Goblin Engineering
UPDATE `npc_trainer` SET `ReqSpell`= 20222 WHERE `SpellID` IN (23486, 36954, 30565, 30566, 12755, 12718, 12908, 9273, 23078, 12717, 12716, 8895, 12758, 30563, 12760, 15628, 30560, 12754, 30558, 12715, 13240, 56514);
```

**Comment by claudiodfc** [comment]
Tested and near perfect 👍🏽 

- Nixx Sprocketspring had 0 recipes that I could learn from due to not having Goblin Engineering.
- Oglethorpe Obnoticus had 1 recipe available called `Inlaid Mithril Cylinder Plans`

File: data/sql/updates/pending_db_world/rev_1638616990639666200.sql (hunk 1 above)
Comment (UltraNix) [reply data/sql/updates/pending_db_world/rev_1638616990639666200.sql:3]: I don't want to do it. If someone re-run that sql file, there will be errors about duplicate column.

**Pull request merged**